presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSource.java

```diff
@@ -186,11 +188,18 @@ public Page getNextPage()
                         continue;
                     }
 
+                    String fieldName = fileSchema.getFields().get(fieldIndex).getName();
                     List<String> path = new ArrayList<>();
-                    path.add(fileSchema.getFields().get(fieldIndex).getName());
+                    path.add(fieldName);
                     if (ROW.equals(type.getTypeSignature().getBase())) {
```

Comment:
Also why don't these have the `getDescriptor(fileSchema, requestedSchema, path).isPresent()` check to determine if the column is in the file?